src/sql/workbench/contrib/charts/browser/chartView.ts

```diff
@@ -228,15 +234,33 @@ export class ChartView extends Disposable implements IPanelView {
 		}
 	}
 
-	private shouldGraph() {
+	private showGraph() {
 		// Check if we have the necessary information
 		if (this._currentData && this._queryRunner) {
 			// check if we are being asked to graph something that is available
 			let batch = this._queryRunner.batchSets[this._currentData.batchId];
 			if (batch) {
 				let summary = batch.resultSetSummaries[this._currentData.resultId];
 				if (summary) {
-					this._queryRunner.getQueryRows(0, summary.rowCount, this._currentData.batchId, this._currentData.resultId).then(d => {
+					const maxRowCount = this._configurationService.getValue<IChartsConfiguration>('charts').maxRowCount;
+					const storageKey = 'charts/ignoreMaxRowCountExceededNotification';
+
+					if (summary.rowCount > maxRowCount) {
+						this._adsTelemetryService.sendTelemetryEvent('charts.maxRowCountExceeded');
```

Comment:
Do we have an event for every time someone charts something too? MIght be good to see what % of the time this happens

(what I'd suggest is that we either add/modify an event that is sent every time someone shows the graph and include a boolean property `maxRowCountExceeded` to make that easier to figure out)